module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -0,0 +1,241 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+class LibGuidesProfile implements
+    RecommendInterface
+{
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Search results object
+     * 
+     * @var \VuFind\Search\Base\Results 
+     */
+    protected $results;
+
+    /**
+     * Map of LibGuides account ID to data about that account
+     *
+     * @var array
+     */
+    protected $idToAccount = [];
+
+    /**
+     * Map of LibGuides subject name (lowercase) to ID of an account
+     * (presumably a librarian) who wrote its subject guide
+     *
+     * @var array
+     */
+    protected $subjectToId = [];
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param array      $config Config object representing LibGuidesAPI.ini
+     * @param HttpClient $client VuFind HTTP client
+     */
+    public function __construct($config, HttpClient $client)
+    {
+        $this->libGuides = new LibGuides(
+            $client,
+            $config->General->api_base_url,
+            $config->General->client_id,
+            $config->General->client_secret
+        );
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search.  This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        // TODO: cache data, check it for staleness
```

Comment:
I do see that VuFind has a nice object cache...planning to use it.

Reply:
@demiankatz Question about caching.  I would like to take advantage of the ttl functionality [built into StorageInterface](https://docs.laminas.dev/laminas-cache/v3/storage/adapter/#basic-configuration-options), but if I use the existing object cache as I think is [recommended](https://vufind.org/wiki/development:architecture:caching), I'd be setting the ttl for everything else using it.  Is that correct?  If so I'd lean to just creating a separate cache, but maybe there's a reason I shouldn't?

Reply:
If you use \VuFind\Cache\CacheTrait, it supports setting different ttl's for different items within a single cache. That's what we do in much of the other code. It might be more elegant to utilize the native built-in ttl, but this mechanism is a little more flexible.

Reply:
Caching implemented.